containers/src/impl/Kokkos_UnorderedMap_impl.hpp

```diff
@@ -27,6 +27,18 @@
 namespace Kokkos {
 namespace Impl {
 
+//! Append to the label contained in view_ctor_prop.
+template <typename... P>
+auto append_to_label(const ViewCtorProp<P...>& view_ctor_prop,
+                     const std::string& label) {
+  using vcp_t = ViewCtorProp<P...>;
+  static_assert(vcp_t::has_label);
+  vcp_t new_ctor_props(view_ctor_prop);
+  static_cast<ViewCtorProp<void, std::string>&>(new_ctor_props)
+      .value.append(label);
+  return new_ctor_props;
+}
+
```

Comment:
just a thought: seems like this function (or variation of it) is being moved around where needed, 
and right here this is only used for unorderedmap. However this function has clearly no ties to unoredered map but just pertain to the ViewCtorProp. What about moving it closer to where ViewCtorProp is defined? It could be that, maybe, using this appending fucntion is something that could potentially be used elsewhere. No idea yet where, but keeping it closer to ViewCtorProp and maybe renaming it makes more sense to me and we don't have to keep moving it around.

Reply:
I generally agree with your suggestion but I would rather have this tackled in some other follow up PR.
Are they other such functions that would need to be relocated?

Reply:
no sure about other functions, i can try to look into it